fnet/network.go

```diff
@@ -160,3 +160,20 @@ func Proxy(port string, dest *net.TCPAddr) *net.TCPAddr {
 func ProxyToDestination(listenPort string, destination string) *net.TCPAddr {
 	return Proxy(listenPort, ResolveDestination(destination))
 }
+
+// Join the input Host And Port in the form of hostname:port, ip:port or :port. And returns
+func JoinHostAndPort(inputPort string, addr *net.TCPAddr) string {
```

Comment:
I checked the caller in uihandler; addr is never nil so the code simplifies to:
```golang
	urlHostPort := addr.String()
	if strings.HasPrefix(inputPort, ":") || !strings.Contains(inputPort, ":") {
            urlHostPort = fmt.Sprintf("localhost:%d", addr.Port)
	}
	return urlHostPort
```

let's rename this function to `NormalizeHostPort` and the comment to
```
// NormalizeHostPort generates host:port string for the address or uses localhost instead of [::] 
// when the original port binding input didn't specify an address
```

ps: the linter should have complained that the comment doesn't match the function